### PR TITLE
SlotMap: Hold all replicas instead of only one.

### PR DESCRIPTION
**Pull request by nihohit**

### CODE REVIEW

File: redis/src/cluster_routing.rs [hunk 1]
```diff
@@ -460,39 +458,36 @@ pub enum SlotAddr {
...
     }
 
     pub(crate) fn slot_addr(&self, slot_addr: SlotAddr) -> &str {
-        if slot_addr == SlotAddr::Replica {
-            self.0[1].as_str()
+        if slot_addr == SlotAddr::Master || self.replicas.is_empty() {
+            self.primary.as_str()
         } else {
-            self.0[0].as_str()
+            self.replicas[0].as_str()
```
Comment (barshaul) [inline redis/src/cluster_routing.rs:475]: Why do we always return the first replica?

File: redis/src/cluster_topology.rs [hunk 2]
```diff
@@ -581,4 +582,85 @@ mod tests {
...
+
+    #[test]
+    fn test_slot_map_get_multi_node() {
+        let slot_map = get_slot_map();
+        let mut addresses =
+            slot_map.addresses_for_multi_routing(&MultipleNodeRoutingInfo::MultiSlot(vec![
+                (Route::new(1, SlotAddr::Master), vec![]),
+                (Route::new(2001, SlotAddr::Replica), vec![]),
+            ]));
+        addresses.sort();
```
Comment (barshaul) [inline redis/src/cluster_topology.rs:658]: Why do we need to sort it?

**Review by barshaul — APPROVED** [review]

File: redis/src/cluster_routing.rs (hunk 1 above)
Comment (nihohit) [reply redis/src/cluster_routing.rs:475]: would you prefer to always get a random replica?

File: redis/src/cluster_topology.rs (hunk 2 above)
Comment (nihohit) [reply redis/src/cluster_topology.rs:658]: for comparisons. We don't promise anything about the order pf returned addresses.

File: redis/src/cluster_routing.rs (hunk 1 above)
Comment (barshaul) [reply redis/src/cluster_routing.rs:475]: yes

File: redis/src/cluster_topology.rs (hunk 2 above)
Comment (nihohit) [reply redis/src/cluster_topology.rs:658]: oops 🤷

**Pull request merged**